src/Collection.php

```diff
@@ -503,62 +480,58 @@ public static function fromGenerator(Generator $generator): self
      */
     public static function fromIterable(iterable $iterable): self
     {
-        return new self(static fn (): Iterator => new IterableIterator($iterable));
+        return new self(static fn (): Generator => yield from new IterableIteratorAggregate($iterable));
     }
 
     /**
-     * @pure
-     *
      * @param resource $resource
      *
      * @return self<int, string>
      */
     public static function fromResource($resource): self
     {
-        return new self(static fn (): IteratorAggregate => new ResourceIteratorAggregate($resource));
+        return new self(static fn (): Generator => yield from new ResourceIteratorAggregate($resource));
     }
 
     /**
-     * @pure
```

Comment:
I just noticed some of these `@pure` annotations are being removed, but not everywhere. I think just in `src/Collection.php`?

How come this is needed?

Reply:
Same comment as the previous one. We are not the only ones... https://github.com/marcosh/lamphpda/blob/master/adr/2022-01-25-pure-callable.md

Reply:
> Same comment as the previous one. We are not the only ones... https://github.com/marcosh/lamphpda/blob/master/adr/2022-01-25-pure-callable.md

I get this one, but in this project we didn't go that far to use `pure-callable` types 😬 .

I think the errors you saw in this project might be related to the iterators project not having things marked as `@pure`.

In any case it might be too much of a hassle to maintain this kind of stuff in PHP, in which case I think it's fine to remove them but we can do it for the whole codebase. You can do it in a separate PR though